apkid/rules/elf/obfuscators.yara

```diff
@@ -371,3 +371,20 @@ rule dexguard_native : obfuscator
       is_elf
       and any of them
 }
+
+rule snapprotect : obfuscator
+{
+  meta:
+    description = "SnapProtect"
+    url         = "https://www.snapchat.com/"
+    sample      = "6dcd634e41304e41b91b49a3c77872a3c7ce28777bab016bd37f79bc7bb08274"
+    author      = "Eduardo Novella"
+
+  strings:
+    $a = "clang version 7.0.0 (snap.protect version 2.4.0 - df15518f469ca4749b08/93d2c161df4b9b202bce)"
+    $b = /clang version \d.\d.\d \(snap.protect version \d.\d.\d/
+    $c = " (snap.protect version "
+
+  condition:
+    is_elf and any of them
```

Comment:
I think it's kind of silly to have such an overlap of definitions.

I suggest removing `$a` and adding as a comment, basically to say "this is the identifier we're looking for, with the below rules"...

`$b` looks find, full overlap with `$a`.

Utilizing `$c` in case `$b` doesn't hit sort of makes sense to me.

For the condition, I'd suggest dropping the `any of them` as you're realistically only requiring 2 things to be true, so something like... `is_elf and 1 of ($a,$b)` will end up creating a faster rule. This is assuming we nix `$a` in the current form and renamed `$b` and `$c`.

Reply:
Good feedback! You know that this rule was written very quickly to grab as many matches as possible :P

Reply:
Totally get it. Just chunking through them per usual :)